runtime/service.go

```diff
@@ -514,14 +521,20 @@ func (s *service) StopVM(requestCtx context.Context, request *proto.StopVMReques
 		return nil, err
 	}
 
-	// The graceful shutdown logic, including stopping the VM, is centralized in Shutdown. We set "Now" to true
-	// to ensure that the service will actually shutdown even if there are still containers being managed.
-	_, err = s.Shutdown(requestCtx, &taskAPI.ShutdownRequest{Now: true})
-	if err != nil {
-		return nil, err
+	for {
```

Comment:
I'm not sure we need the `for` statement wrapping the `select`, unless I'm missing something we should be good with just the `select` here

Reply:
Actually, I took a second look and think we will need a little bit more than just removing the `for`. We want to make sure that if we hit the timeout while blocked on the `Shutdown` call, the `StopVM` method doesn't stay blocked. I think the simplest way to do this will be running `Shutdown` in its own goroutine. Roughly sketched out:
```
shutdownCh := make(chan error)
go func() {
    defer close(shutdownCh)
    _, err = s.Shutdown(...)
    shutdownCh <- err
}()

select {
    case <- ctx.Done():
        ...
    case err = <-shutdownCh:
        if err != nil {
            return nil, err
        }
        return Empty{}, nil
}
```